SingularityUI/app/controllers/Tail.coffee

```diff
@@ -16,11 +16,14 @@ class TailController extends Controller
             model: @models.taskHistory
 
         app.showView @view
+
         if @offset?
-            @collections.logLines.fetchOffset @offset
+            $.when( @models.taskHistory.fetch(), @collections.logLines.fetchOffset(@offset) ).then =>
+                @view.afterInitialData()
         else
-            @collections.logLines.fetchInitialData()
-
+            $.when( @models.taskHistory.fetch(), @collections.logLines.fetchInitialData() ).then =>
+                @view.afterInitialData()
+        
         @refresh()
```

Comment:
ah right, didn't mean to still call @refresh again there. i like the idea of storing the fetchOffset / fetchInitialData result, but just realized I need to call a different method if the link is an offset (so it doesn't tail), so I removed the additional refresh call.